packages/components/src/components/select/select.ts

```diff
@@ -629,53 +760,105 @@ export default class SdSelect extends SolidElement implements SolidFormControl {
-    const hasLabel = this.label ? true : !!hasLabelSlot;
-    const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
+    // Slots
+    const slots = {
+      default: this.hasSlotController.test('[default]'),
+      label: this.hasSlotController.test('label'),
+      clearIcon: this.hasSlotController.test('clear-icon'),
+      expandIcon: this.hasSlotController.test('expand-icon'),
+      helpText: this.hasSlotController.test('help-text')
+    };
+
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
     const hasClearIcon = this.clearable && !this.disabled && this.value.length > 0;
-    const isPlaceholderVisible = this.placeholder && this.value.length === 0;
 
+    const isInvalid = this.hasAttribute('data-user-invalid') && this.required && !this.checkValidity();
+    const isValid = this.hasAttribute('data-user-valid') && this.required && this.checkValidity();
+
+    // Hierarchy of input states:
+    const selectState = this.disabled
+      ? 'disabled'
+      : this.hasFocus && isInvalid
+        ? 'activeInvalid'
+        : this.hasFocus && isValid
+          ? 'activeValid'
+          : this.hasFocus || this.open
+            ? 'active'
+            : isInvalid
+              ? 'invalid'
+              : isValid
+                ? 'valid'
+                : 'default';
+
+    // Conditional Styles
+    const cursorStyles = this.disabled ? 'cursor-not-allowed' : 'cursor-pointer';
+
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[selectState];
+
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[selectState];
+
+    const iconMarginLeft = { sm: 'ml-1', md: 'ml-2', lg: 'ml-2' }[this.size];
+    const iconSize = {
+      sm: 'text-base',
+      md: 'text-lg',
+      lg: 'text-xl'
+    }[this.size];
+
+    // Template
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx('form-control relative text-left', cursorStyles, textColor, textSize, this.open && 'z-10')}
```

Comment:
If you are using the textColor only once, I would suggest to put it directly in the class (for better readability)

```
{
  disabled: 'text-neutral-500',
  readonly: 'text-black',
  activeInvalid: 'text-error',
  activeValid: 'text-success',
  active: 'text-black',
  invalid: 'text-error',
  valid: 'text-success',
  default: 'text-black'
}[selectState];
```

Reply:
Also here:
```suggestion
        class=${cx(
          'form-control relative text-left', cursorStyles, this.open && 'z-10',
          this.disabled ? 'cursor-not-allowed' : 'cursor-pointer',
          this.size === 'sm' ? 'text-sm' : 'text-base',
          {
            disabled: 'text-neutral-500',
            readonly: 'text-black',
            activeInvalid: 'text-error',
            activeValid: 'text-success',
            active: 'text-black',
            invalid: 'text-error',
            valid: 'text-success',
            default: 'text-black'
          }[selectState]
        )}
```

Reply:
It's just my opinion, but I think it's easier to read this way.